diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache mpi-serial-MPIserial_2.3.0.orig/Makefile mpi-serial-MPIserial_2.3.0/Makefile
--- mpi-serial-MPIserial_2.3.0.orig/Makefile	2021-01-03 04:34:49.000000000 +0100
+++ mpi-serial-MPIserial_2.3.0/Makefile	2023-07-06 17:20:42.561618054 +0200
@@ -83,11 +83,9 @@
 tests:
 	cd tests; make
 
-install: lib
+install: $(LIB)
 	$(MKINSTALLDIRS) $(libdir) $(includedir)
 	$(INSTALL) lib$(MODULE).a -m 644 $(libdir)
 	$(INSTALL) mpi.h -m 644 $(includedir)
 	$(INSTALL) mpif.h -m 644 $(includedir)
-
-
-
+	$(INSTALL) mpi.mod -m 644 $(includedir)
