diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache mpi-serial-MPIserial_2.3.0.types/ic_merge.c mpi-serial-MPIserial_2.3.0.fix-intercomm-merge/ic_merge.c
--- mpi-serial-MPIserial_2.3.0.types/ic_merge.c	2023-07-06 17:31:16.285643612 +0200
+++ mpi-serial-MPIserial_2.3.0.fix-intercomm-merge/ic_merge.c	2023-07-07 09:24:08.111183751 +0200
@@ -10,6 +10,6 @@
 
 int MPI_Intercomm_merge( MPI_Comm intercomm, int high, MPI_Comm *newintracomm )
 {
-  newintracomm = (MPI_Comm *)intercomm;
+  *newintracomm = intercomm;
   return(MPI_SUCCESS);
 }
